### PR TITLE
[DO NOT MERGE] DQI: Add command to evaluate a list of quality criteria for products

### PR DESCRIPTION
**Pull request by olivierpontier**
**Description (for Contributor and Core Developer)**

:warning:  This is experimentation and needs to be discussed with the team
In order to measure the performances and ease the debugging of the data quality evaluation, this pull request provides a new CLI command for developers that allows running the data quality evaluation only on the criteria they choose (without heavy process).

Example of use:
```
APP_ENV=dev docker-compose run --rm -u www-data php bin/console pim:data-quality-insights:evaluate-criteria 1207 -t product -c enrichment_image -c completeness_of_required_attributes

```

image

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Application/ProductEvaluation/EvaluateCriteria.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        private GetEvaluableProductValuesQueryInterface $getEvaluableProductValuesQuery,
+        private LoggerInterface $logger,
+        private string $productType
+    ) {
+    }
+
+    /**
+     * @param string[] $criteriaCodes
+     */
+    public function __invoke(ProductIdCollection $productIdCollection, array $criteriaCodes, callable $afterEvaluateCriterion = null): void
```
Comment (LaurentPetard) [inline src/Akeneo/Pim/Automation/DataQualityInsights/back/Application/ProductEvaluation/EvaluateCriteria.php:38]: I don't think that it's a good practice to have a callable as parameter in a public method. It potentially allow to do anything

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Application/ProductEvaluation/EvaluateCriteria.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+use Akeneo\Pim\Automation\DataQualityInsights\Domain\ValueObject\ProductId;
+use Akeneo\Pim\Automation\DataQualityInsights\Domain\ValueObject\ProductIdCollection;
+use Psr\Log\LoggerInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class EvaluateCriteria
```
Comment (LaurentPetard) [inline src/Akeneo/Pim/Automation/DataQualityInsights/back/Application/ProductEvaluation/EvaluateCriteria.php:22]: I don't understand what is the functional need of this service ? It seems that it's only reason to be is to be used with a CLI command by a developer.
Maybe it should be less confusing to place this service in `Infrastructure` near the Symfony command that use this service ?  Or simply in private methods inside the command ?

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Command/EvaluateProductCriterionCommand.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class EvaluateProductCriterionCommand extends Command
```
Comment (LaurentPetard) [inline src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Command/EvaluateProductCriterionCommand.php:22]: For what purpose do we need this command ? Is it a part of the work to do for PLG-809 ? 
won't it bring even more confusion that today with the commands `pim:data-quality-insights:evaluations`, `pim:data-quality-insights:initialize-products-evaluations`, `pim:data-quality-insights:prepare-evaluations`, `pim:data-quality-insights:evaluate-all-products` ?

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Application/ProductEvaluation/EvaluateCriteria.php (hunk 1 above)
Comment (olivierpontier) [reply src/Akeneo/Pim/Automation/DataQualityInsights/back/Application/ProductEvaluation/EvaluateCriteria.php:38]: You are right, the callable signature could be replaced by an invokable interface

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Command/EvaluateProductCriterionCommand.php (hunk 3 above)
Comment (olivierpontier) [reply src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Command/EvaluateProductCriterionCommand.php:22]: Good point. It is not related to the work on rationalizing the CLI commands of the Data QUality Insights bounded context. It is only a proposition of tool in order to ease the development around the Data quality evaluation.

Comment (FSAkeneo) [reply src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Command/EvaluateProductCriterionCommand.php:22]: CriterionEvaluationMeasurementCommand ?

**Pull request closed without merging**